tfe/provider.go

```diff
@@ -282,36 +282,70 @@ func getClient(tfeHost, token string, insecure bool) (*tfe.Client, error) {
 // cliConfig tries to find and parse the configuration of the Terraform CLI.
 // This is an optional step, so any errors are ignored.
 func cliConfig() *Config {
-	config := &Config{}
-
-	// Detect the CLI config file path.
-	configFilePath := os.Getenv("TERRAFORM_CONFIG")
+	mainConfig := &Config{}
+	combinedConfig := &Config{}
+
+	// There might be credentials in the main CLI config file (manually entered)
+	// AND/OR the credentials file (auto-configured by terraform login), so we
+	// need to consult both. As per the behavior of Terraform itself, the main
+	// config file wins if both sources have credentials for a given host.
+
+	// To find the main CLI config file, follow Terraform's own logic: try
+	// TF_CLI_CONFIG_FILE, then try TERRAFORM_CONFIG, then try the default
+	// location.
+	configFilePath := os.Getenv("TF_CLI_CONFIG_FILE")
+	if configFilePath == "" {
+		configFilePath = os.Getenv("TERRAFORM_CONFIG")
+	}
```

Comment:
What do you think about refactoring this a tad bit for readability? I added an example here https://github.com/hashicorp/terraform-provider-tfe/commit/ad40849a148691621f8ac8b75483b475dff60e2e